File: app/src/main/java/team/triplog/presentation/activity/MainActivity.java
package team.triplog.presentation.activity;

import android.os.Bundle;
import android.widget.FrameLayout;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import com.google.android.material.bottomnavigation.BottomNavigationView;

import team.triplog.R;
import team.triplog.data.local.User;
import team.triplog.presentation.fragment.MainHomeFragment;
import team.triplog.presentation.fragment.MainMoreFragment;
import team.triplog.presentation.fragment.MainSearchFragment;
import team.triplog.presentation.fragment.MainTripFragment;

public class MainActivity extends BaseActivity {

    FrameLayout fragmentMain;
    BottomNavigationView bottomNavigation;
    User user;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        init();
        setData();
        setUi();
    }

    private void init() {
        fragmentMain = findViewById(R.id.fragment_main);
        bottomNavigation = findViewById(R.id.bottom_navigation);

        bottomNavigation.setOnNavigationItemSelectedListener(onNavigationItemSelectedListener);
    }

    private void setData() {
        user = realm.where(User.class).findFirst();
    }

    private void setUi() {
        fragmentChange(new MainHomeFragment(user));
    }

    private void fragmentChange(Fragment fragment) {
        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        fragmentTransaction.replace(R.id.fragment_main, fragment);
        fragmentTransaction.commit();
    }

    BottomNavigationView.OnNavigationItemSelectedListener onNavigationItemSelectedListener =
            menuItem -> {
                menuItem.setChecked(true);
                switch (menuItem.getItemId()) {
                    case R.id.navigation_home:
                        fragmentChange(new MainHomeFragment(user));
                        break;
                    case R.id.navigation_trip:
                        fragmentChange(new MainTripFragment());
                        break;
                    case R.id.navigation_search:
                        fragmentChange(new MainSearchFragment());
                        break;
                    case R.id.navigation_more:
                        fragmentChange(new MainMoreFragment(realm, user));
                        break;
                }
                return false;
            };
}

File: app/src/main/java/team/triplog/presentation/main/activity/MainActivity.kt
package team.triplog.presentation.main.activity

import android.os.Bundle
import android.view.MenuItem
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import com.google.android.material.bottomnavigation.BottomNavigationView
import kotlinx.android.synthetic.main.activity_main.*
import org.koin.androidx.viewmodel.ext.android.viewModel
import team.triplog.R
import team.triplog.databinding.ActivityMainBinding
import team.triplog.presentation.base.BaseActivity
import team.triplog.presentation.fragment.MainSearchFragment
import team.triplog.presentation.fragment.MainTripFragment
import team.triplog.presentation.main.fragment.MainHomeFragment
import team.triplog.presentation.main.fragment.MainMoreFragment
import team.triplog.presentation.main.viewmodel.MainViewModel


class MainActivity : BaseActivity() {
    private lateinit var binding: ActivityMainBinding
    private val viewModel: MainViewModel by viewModel()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        binding.apply {
            lifecycleOwner = this@MainActivity
            viewModel = this@MainActivity.viewModel
        }

        init()
        addObservableData()
    }

    private fun init() {
        bottom_navigation.setOnNavigationItemSelectedListener(onNavigationItemSelectedListener)
    }

    private fun addObservableData() {
        fragmentChange(MainHomeFragment())

    }

    private fun fragmentChange(fragment: Fragment) {
        val fragmentTransaction = supportFragmentManager.beginTransaction()
        fragmentTransaction.replace(R.id.fragment_main, fragment)
        fragmentTransaction.commit()
    }


    private var onNavigationItemSelectedListener =
        BottomNavigationView.OnNavigationItemSelectedListener { menuItem: MenuItem ->
            menuItem.isChecked = true
            when (menuItem.itemId) {
                R.id.navigation_home -> fragmentChange(MainHomeFragment())
                R.id.navigation_trip -> fragmentChange(MainTripFragment())
                R.id.navigation_search -> fragmentChange(MainSearchFragment())
                R.id.navigation_more -> fragmentChange(MainMoreFragment(viewModel.readUser("").value))
            }
            false
        }
}

